### PR TITLE
Kafka module

### PR DESCRIPTION
**Pull request by Spring3**
Kafka plugin for logtify logger

## Usage
Used with [logtify](https://github.com/dial-once/node-logtify) module.

```js
require('logtify-kafka')({ KAFKA_HOST: 'kafka-broker.io:9092', KAFKA_TOPIC: 'example' });
const { stream, logger } = require('logtify')();
logger.log('error', new Error('Test error'));
logger.info('Hello world!');
```

The subscriber will send the given data to Kafka broker if:
* ``message.level >= ('MIN_LOG_LEVEL_KAFKA' || 'MIN_LOG_LEVEL')``
* ``process.env.KAFKA_LOGGING === 'true' || settings.KAFKA_LOGGING === true``
* if not given, process.env.KAFKA_LOGGING is considered `true` by default

**Settings**:
Module can be configured by both env variables or config object. However, env variables have a higher priority.
```js
{
  KAFKA_HOST: 'kafka-broker:9092',
  KAFKA_TOPIC: 'example', // the plugin will attempt to create this topic on start
  KAFKA_LOGGING: true|false, // true by default
  KAFKA_CONNECT_TIMEOUT: 5000,
  KAFKA_REQUEST_TIMEOUT: 10000,
  MIN_LOG_LEVEL_KAFKA: 'silly|verbose|info|warn|error',
  LOG_TIMESTAMP: 'true',
  LOG_ENVIRONMENT: 'true',
  LOG_LEVEL: 'true',
  LOG_REQID: 'true', // only included when provided with metadata
  LOG_CALLER_PREFIX: 'true' // additional prefix with info about caller module/project/function
}
```

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  constructor(configs) {
+    super();
+    this.ready = false;
+    this.settings = configs || {};
+    this.kafkaClient = undefined;
+    this.kafkaProducer = undefined;
+    if (this.settings.KAFKA_HOST && this.settings.KAFKA_TOPIC) {
+      this.initialize();
+    } else {
+      console.warn('Kafka logtify module is not active due to a missing KAFKA_HOST and/or KAFKA_TOPIC');
```
Comment (mrister) [inline src/index.js]: would it be wise to use logtify (with , also, console) to log this? like here https://github.com/dial-once/node-logtify-kafka/pull/1/files#diff-1fdf421c05c1140f6d71444ea2b27638R52

**Review by mrister — APPROVED** [review]

File: src/index.js (hunk 1 above)
Comment (Spring3) [reply src/index.js]: Good suggestion. Applied

**Review by ofkindness — APPROVED** [review]
I think `kafka-node` is not the best choice as a basic module. Plus they were updated recently to v3.0.1 (and you using v2.6.1). The `node-rdkafka` module looks much better from my experience

**Comment by Spring3** [comment]
@ofkindness unfortunately when I tried to install a `node-rdkafka` module, node-gyp failed to build it. That's why I decided to use `kafka-node` instead. Thanks for the notice about the version. I will update this project

**Review by ykhatal — APPROVED** [review]

**Review by jkernech — APPROVED** [review]

**Pull request merged**